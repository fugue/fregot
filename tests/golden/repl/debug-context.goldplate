{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file", "debug-context.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"stdout": "${GOLDPLATE_NAME}.stdout"}
    ],
    "stdin": [
        ":break debug_context.valid_security_groups",
        "data.debug_context.test_valid_security_groups",
        "security_groups"
    ]
}
