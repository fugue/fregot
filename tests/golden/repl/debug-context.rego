# Check that we can access the right things from the debug context.  In
# particular, the input and rules form the "current" package.
package debug_context

security_groups[id] = security_group {
  security_group = input.resources[id]
  security_group.type = "security_group"
}

valid_security_groups[id] = security_group {
  security_group = security_groups[id]
  security_group.good_boy
}

test_valid_security_groups {
  valids = valid_security_groups with input as {
    "resources": [
      {"id": "1", "type": "security_group", "good_boy": true},
      {"id": "2", "type": "security_group", "good_boy": false},
      {"id": "3", "type": "virtual_machine"}
    ]
  }
}
