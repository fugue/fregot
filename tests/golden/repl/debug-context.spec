{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file", "debug-context.rego"],
    "asserts": [
        {"exit_code": 0},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"stdout": "${SPEC_NAME}.stdout"}
    ],
    "stdin": [
        ":break debug_context.valid_security_groups",
        "data.debug_context.test_valid_security_groups",
        "security_groups"
    ]
}
