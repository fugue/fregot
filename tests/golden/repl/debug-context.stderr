F u g u e   R E G O   T o o l k i t
fregot v0.11.1 repl - use :help for usage info
Set breakpoint at debug_context.valid_security_groups
